code/game/machinery/metaldetector.dm

```diff
@@ -170,35 +170,13 @@
 
 	if(usr) usr.set_machine(src)
 
-	if (href_list["idmode"])
-
-		if(idmode)
-			idmode = 0
-			//
-
-		else
-			idmode = 1
-
-
-
-	if (href_list["scanmode"])
-
-		if(scanmode)
-			scanmode = 0
-			//
-
-		else
-			scanmode = 1
-
-
-	if (href_list["senmode"])
-
-		if(senset)
-			senset = 0
-			//
-
-		else
-			senset = 1
+	switch(href_list["action"])
+		if("idmode")
+			idmode = !idmode
+		if("scanmode")
+			scanmode = !scanmode
+		if("senmode")
+			senset = !senset
```

Comment:
Where it's doing if(..()) return, it should be retuning 1, not null, because if any childs of the VALID were to be made sanity checks on topic for that child would break